control-plane/pkg/core/config/utils_test.go

```diff
@@ -346,3 +349,133 @@ func TestEgressConfigFromDelivery(t *testing.T) {
 		})
 	}
 }
+
+func TestMergeEgressConfig(t *testing.T) {
+
+	tt := []struct {
+		name     string
+		e0       *contract.EgressConfig
+		e1       *contract.EgressConfig
+		expected *contract.EgressConfig
+	}{
+		{
+			name: "e0 nil",
+			e1: &contract.EgressConfig{
+				Retry: 42,
+			},
+			expected: &contract.EgressConfig{
+				Retry: 42,
+			},
+		},
+		{
+			name: "e1 nil",
+			e0: &contract.EgressConfig{
+				Retry: 42,
+			},
+			expected: &contract.EgressConfig{
+				Retry: 42,
+			},
+		},
+		{
+			name: "e0 retry priority",
+			e0: &contract.EgressConfig{
+				Retry: 42,
+			},
+			e1: &contract.EgressConfig{
+				Retry: 43,
+			},
+			expected: &contract.EgressConfig{
+				Retry: 42,
+			},
+		},
+		{
+			name: "e0 dead letter priority",
+			e0: &contract.EgressConfig{
+				DeadLetter: "e0",
+			},
+			e1: &contract.EgressConfig{
+				Retry:      43,
+				DeadLetter: "e1",
+			},
+			expected: &contract.EgressConfig{
+				DeadLetter: "e0",
+				Retry:      43,
+			},
+		},
+		{
+			name: "e0 timeout priority",
+			e0: &contract.EgressConfig{
+				DeadLetter: "e0",
+				Timeout:    100,
+			},
+			e1: &contract.EgressConfig{
+				Retry:      43,
+				DeadLetter: "e1",
+				Timeout:    101,
+			},
+			expected: &contract.EgressConfig{
+				DeadLetter: "e0",
+				Retry:      43,
+				Timeout:    100,
+			},
+		},
+		{
+			name: "e0 backoff delay priority",
+			e0: &contract.EgressConfig{
+				DeadLetter:   "e0",
+				Timeout:      100,
+				BackoffDelay: 4001,
+			},
+			e1: &contract.EgressConfig{
+				Retry:        43,
+				DeadLetter:   "e1",
+				Timeout:      101,
+				BackoffDelay: 4000,
+			},
+			expected: &contract.EgressConfig{
+				DeadLetter:   "e0",
+				Retry:        43,
+				Timeout:      100,
+				BackoffDelay: 4001,
+			},
+		},
+		{
+			name: "e0 backoff policy priority",
+			e0: &contract.EgressConfig{
+				DeadLetter:    "e0",
```

Comment:
what if this has `Retry`?

Reply:
it should have prio over `e1`, right ?

Reply:
Right, let me add another test with the retry parameter

Reply:
Done!